### DISCUSSION TITLE
`Transform.inv` seems incorrect

### DISCUSSION THREAD

**Discussion by simon-bachhuber** [google/brax#310] [General]
Dear Brax authors,

while scrolling through some of the source i stumpled upon

```python
class Transform(_Base):
  def inv(self):
    """Invert the transform."""
    return Transform(pos=-1.0 * self.pos, rot=math.quat_inv(self.rot))
```

Is this really correct? Is it due to some convention i am unaware of?

It also seems inconsistent with Roy Featherstone 
![Screenshot from 2023-01-23 14-30-09](https://user-images.githubusercontent.com/61733064/214051820-9f36b46b-1361-4cf9-9325-8be71bc69ee9.png)

Greetings

**Comment by btaba** [comment]
Hi @SimiPixel we follow a similar convention to [TDS](https://github.com/erwincoumans/tiny-differentiable-simulator/blob/24c1d2a820e4e8a46d2036fe206340615085c8ef/src/math/transform.hpp#L13), where `Transform` is expressed as a translation and quaternion rotation, rather than a 6x6

**Reply by btaba** [reply]
Hi @SimiPixel , indeed thanks so much for pointing this out!! inv is clearly wrong, it should be:
```
rot = math.quat_inv(self.rot)
Transform(pos=math.rotate(-1.0 * self.pos, rot), rot=rot)
```
as in https://github.com/erwincoumans/tiny-differentiable-simulator/blob/master/src/math/transform.hpp#L191-L195

where we are also using right associative transforms (hence the slight diff with the screenshot you provided)

**Reply by simon-bachhuber** [reply]
No worries. As long as 

```python
t = Transform()
t.inv().do(t) == t.do(t.inv()) == Transform.zero()
```

i am convinced :+1:
